File: LeetCodeSolutions/longestvalidparanthesis.py
class Solution(object):
    def longestValidParentheses(self, s):
      
        stack = [-1]
        res = 0
        for i in range(len(s)):
            if s[i] == "(":
                stack.append(i)
            else:
                stack.pop()
                
                if stack:
                    res = max(res, i - stack[-1])
                else:
                    stack.append(i)
                    
        return res

File: LeetCodeSolutions/longestvalidparanthesis.cpp
class Solution
{
public:
   int longestValidParentheses(string s) 
   {
	int MAX = 0; 
	stack<int> stk;
	stk.push(-1);  
	for(int i = 0; i < size(s); i++)
    {if(s[i] == '(') stk.push(i);            
		else{                
			stk.pop();
			if(stk.empty()) stk.push(i);
			else MAX = max(MAX, i - stk.top());
		}  
    }
	return MAX;
}
};
